### PR TITLE
i18n: Allow to pass extra arguments to msgfmt

### PR DESCRIPTION
**Pull request by sardemff7**
I need that to pass extra `--keyword` for my desktop format files.

### CODE REVIEW

File: mesonbuild/modules/i18n.py [hunk 1]
```diff
@@ -62,6 +62,10 @@ def merge_file(self, state, args, kwargs):
 
         kwargs['command'] = ['msgfmt', '--' + file_type,
                              '--template', '@INPUT@', '-d', podir, '-o', '@OUTPUT@']
+
+        extra_args = mesonlib.stringlistify(kwargs.get('args', []))
+        kwargs['command'].append(extra_args) if extra_args
```
Comment (ignatenkobrain) [inline mesonbuild/modules/i18n.py]: __[flake8]__
* [E999] SyntaxError: invalid syntax

Comment (sardemff7) [inline mesonbuild/modules/i18n.py]: Oops, first time in Python, I guess I copied wrong (well, there’s `else None` in the part I copied but here it seemed wrong)

File: mesonbuild/modules/i18n.py [hunk 2]
```diff
@@ -62,6 +62,11 @@ def merge_file(self, state, args, kwargs):
 
         kwargs['command'] = ['msgfmt', '--' + file_type,
                              '--template', '@INPUT@', '-d', podir, '-o', '@OUTPUT@']
+
+        extra_args = mesonlib.stringlistify(kwargs.pop('args', []))
+        if extra_args is not None:
```
Comment (QuLogic) [inline mesonbuild/modules/i18n.py]: I think the preference in meson code is more like: `if 'args' in kwargs: do stuff`.

Comment (TingPing) [inline mesonbuild/modules/i18n.py]: or just `if extra_args`. The `None` check is certainly wrong since it would be `[]`.

Comment (sardemff7) [inline mesonbuild/modules/i18n.py]: I’ll go with the first

Comment (TingPing) [inline mesonbuild/modules/i18n.py]: Well you have to check for an empty list no matter.

Comment (QuLogic) [inline mesonbuild/modules/i18n.py]: Actually, shouldn't it be `extend`? And then an empty list wouldn't matter.

**Comment by jpakkane** [comment]
This message is posted to every outstanding merge request.

We have transitioned from Github wiki to our new documentation system that generates all documentation directly from the Git repo. This means that from now on every merge request must come with corresponding documentation changes.

If your MR requires documentation changes, do the necessary changes to files in `docs/markdown`.

If your MR adds a new feature, add a description to the release notes in `docs/markdown/Release-notes-for-0.41.0.md`.

If this commit is just a bugfix with no functional changes, you don't need to do anything.

Thank you

File: docs/markdown/i18n-module.md [hunk 3]
```diff
@@ -26,5 +26,6 @@ This merges translations into a text file using `msgfmt`. See [custom_target](ht
 
 * `po_dir`: directory containing translations, relative to current directory
 * `type`: type of file, valid options are `'xml'` (default) and `'desktop'`
+* `args`: list of extra arguments to pass to `msgfmt`
```
Comment (jpakkane) [inline docs/markdown/i18n-module.md]: This should mention that it is only available since 0.41.0 (doc updates go live upon merging)

**Comment by TingPing** [comment]
To be consistent with other modules it should be `extra_args` not `args`.

**Comment by jpakkane** [comment]
`extra_args` is more exact, let's standardise on that.

File: docs/markdown/i18n-module.md [hunk 4]
```diff
@@ -26,5 +26,6 @@ This merges translations into a text file using `msgfmt`. See [custom_target](ht
 
 * `po_dir`: directory containing translations, relative to current directory
 * `type`: type of file, valid options are `'xml'` (default) and `'desktop'`
+* `extra_args`: (*Added 0.41.0*) list of extra arguments to pass to `msgfmt
```
Comment (jpakkane) [inline docs/markdown/i18n-module.md]: There should be a backtick at the end of this line to close the style block.

Comment (sardemff7) [inline docs/markdown/i18n-module.md]: Oops, my bad.

**Comment by TingPing** [comment]
There should be some tests for this, I don't think after that rebase it even works as-is.

**Comment by sardemff7** [comment]
True, now it works again.

**Comment by jpakkane** [comment]
Still needs a test, though...

**Comment by sardemff7** [comment]
What kind of test? Just testing the called command works? Or testing what command is called (checking meson output)? Testing a full `.merge_file()` result (with a file similar to mine that doesn’t work with defaults)?

**Comment by jpakkane** [comment]
Testing that the extra arg is used is the important bit here. Custom targets sadly do not go to the compilation database, so just checking that does not work. Is there a simple way to do a unit test that uses that extra arg and then check the output?

**Comment by sardemff7** [comment]
Yes, this is easy to do. However, trying to add that to the existing gettext test, I just found that the current test is testing half the features (test.desktop is not translated at all), and that you are lacking some really important features (#1733 and #1739).
Until we have them properly fixed, I can workaround that with different subdirs in the test, or just a different test. Which way should I go?

**Comment by jpakkane** [comment]
Different subdir seems the simplest. Just have a comment in the meson file explaining why it's there.

**Comment by jpakkane** [comment]
ping

**Comment by sardemff7** [comment]
Do.we want to improve this method or just rely on #1757 to provide the feature?

**Comment by TingPing** [comment]
This still just needs tests and otherwise seems fine. I don't think we should wait for a complete replacement.

**Comment by nirbheek** [comment]
> Until we have them properly fixed, I can workaround that with different subdirs in the test, or just a different test. Which way should I go?

Whichever seems easier, please do that and we'll get this in.

**Comment by jpakkane** [comment]
Ping.

**Comment by jpakkane** [comment]
Ping again.

**Comment by sardemff7** [comment]
Honestly I’ve switched to #1757 because it allows more use cases. It’s been a year and a half since I’ve last touched this code, so I’ll just drop this PR and hope the other one will finally get a design settled on so that I can try to gather enough motivation to finish it.

**Pull request closed without merging**